packages/itwin/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx

```diff
@@ -40,64 +29,32 @@ export class TreeWidgetUiItemsProvider implements UiItemsProvider {
     location: StagePanelLocation,
     section?: StagePanelSection
   ): ReadonlyArray<Widget> {
-    const widgets: Widget[] = [];
     const preferredLocation = this._treeWidgetOptions?.defaultPanelLocation ?? StagePanelLocation.Right;
     const preferredPanelSection = this._treeWidgetOptions?.defaultPanelSection ?? StagePanelSection.Start;
-    if (
-      location === preferredLocation &&
-      section === preferredPanelSection &&
-      stageUsage === StageUsage.General
-    ) {
-      const trees: SelectableContentDefinition[] = [];
 
-      if (!this._treeWidgetOptions?.hideTrees?.modelsTree) {
-        trees.push({
-          label: TreeWidget.translate("models"),
-          id: ModelsTreeComponent.Id,
-          render: () => (
-            <ModelsTreeComponent
-              enableElementsClassGrouping={
-                this._treeWidgetOptions?.enableElementsClassGrouping
-                  ? ClassGroupingOption.YesWithCounts
-                  : ClassGroupingOption.No
-              }
-              {...this._treeWidgetOptions?.modelsTreeProps}
-            />
-          ),
-        });
-      }
-
-      if (!this._treeWidgetOptions?.hideTrees?.categoriesTree) {
-        trees.push({
-          label: TreeWidget.translate("categories"),
-          id: CategoriesTreeComponent.Id,
-          render: () => (
-            <CategoriesTreeComponent {...this._treeWidgetOptions?.categoriesTreeProps} />
-          ),
-        });
-      }
-
-      if (this._treeWidgetOptions?.additionalTrees) {
-        trees.push(...this._treeWidgetOptions.additionalTrees);
-      }
-
-      if (this._treeWidgetOptions?.defaultTreeId && trees.length !== 0) {
-        // Adding the defaultTree to first index
-        const { defaultTreeId } = this._treeWidgetOptions;
-        const extractedDefaultTree = trees.filter((tree) => tree.id === defaultTreeId)[0];
-        const index = trees.indexOf(extractedDefaultTree);
-        trees.unshift(trees.splice(index, 1)[0]);
-      }
-
-      widgets.push({
-        id: TreeWidgetId,
-        label: TreeWidget.translate("treeview"),
-        content: <TreeWidgetComponent trees={trees} />,
-        icon: "icon-hierarchy-tree",
-        priority: this._treeWidgetOptions?.defaultTreeWidgetPriority,
-      });
+    if (location !== preferredLocation || section !== preferredPanelSection || stageUsage !== StageUsage.General) {
+      return [];
     }
 
-    return widgets;
+    const trees: TreeDefinition[] = this._treeWidgetOptions?.trees ?? [
```

Comment:
So we have no way to add additional trees? its either just these 2 or we provide everything?

Reply:
These 2 trees are shown by default only for the simplest case when you want to see default `ModelsTree` and `CategoriesTree`. If you want to customize content shown in `TreeWidget` (hide, add, reorder, modify trees) you will need to explicitly specify what you want to see. I know this adds some additional code when you want to have default `ModelsTree` and `CategoriesTree` with some additional trees (about 6 lines) but it makes API more flexible and cleaner.

Reply:
@jason-crow, the goal here is to simplify the API and make it more flexible at the same time.

- The way available trees was controlled was really unintuitive - that was controlled through `hideTrees`, `additionalTrees` and `defaultTreeId` props. To change the order of the trees you would've had to hide all trees and provide them in correct order through `additionalTrees`.

- Configuring default trees - the props were provided through `enableElementsClassGrouping`, `modelsTreeProps` and `categoriesTreeProps`.
  - What's the meaning of `{ modelsTreeProps: { ...customProps }, hideTrees: { modelsTree } }`?
  - What's the meaning of `{ enableElementsClassGrouping: true, modelsTreeProps: { enableElementsClassGrouping: ClassGroupingOption.No }`?

The current approach solves all those inconsistencies - you pass in all the trees you want to see in the widget, in the order you want, with the configuration you want. 

Please let us know if you have any concerns - all 1.0 changes are being pushed to a separate branch and we can make improvements before we release, if necessary.